### PR TITLE
OSS acknowledgements

### PR DESCRIPTION
**Pull request by DrJosh9000**
This adds a compilation of all the licenses and notices of dependencies used to build the agent (ACKNOWLEDGEMENTS.md), and a script to produce the compilation.

I've also made a compressed version, embedded it into clicommand, and added a new subcommand for printing it. This will add around 20KB to the binary.

### CODE REVIEW

File: scripts/generate-acknowledgements.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+fi
+
+OUTPUT_FILE="ACKNOWLEDGEMENTS.md"
+
+# Ensure modules are downloaded
+go mod download
+
+# Get go-licenses tool
+go install github.com/google/go-licenses@latest
+GO_LICENSES="${HOME}/go/bin/go-licenses"
```
Comment (triarius) [inline scripts/generate-acknowledgements.sh:16]: Some people might have installed this by some other means. Some, not necessarily the same, people might also have a different GOPATH.

```suggestion
if ! command -v go-licenses >/dev/null; then
	go install github.com/google/go-licenses@latest
	GO_LICENSES="$(go env GOPATH)/bin/go-licenses"
else
	GO_LICENSES="$(command -v go-licences)"
fi
```

**Review by triarius — COMMENTED** [review]
This is great! I think it is a good idea to have a CI step that generates this and checks there's no diff from what's output by the built binary.

File: scripts/generate-acknowledgements.sh (hunk 1 above)
Comment (triarius) [reply scripts/generate-acknowledgements.sh:16]: This is all the testing I've done:
```
if ! command -v go-licenses >/dev/null; then    go install github.com/google/go-licenses@latest;   GO_LICENSES="$(go env GOPATH)/bin/go-licenses"; else    GO_LICENSES="$(command -v go-licenses)"; fi
[narthana@quirinius ~]$ echo $GO_LICENSES
```

**Review by triarius — APPROVED** [review]
Thanks!

**Comment by DrJosh9000** [comment]
> This is great! I think it is a good idea to have a CI step that generates this and checks there's no diff from what's output by the built binary.

I wonder how often Dependabot will be blocked by it. The probability is probably pretty low that a dependency will change licenses, but it could happen. Dependencies could also add or change notice files.

**Comment by DrJosh9000** [comment]
Actually, that's probably a good thing. If a dependency changes licenses, right now we probably wouldn't notice!

I'll follow up with another PR to add the CI step, in order for it to have a file to diff against.

**Pull request merged**